Data/Graph/Inductive/PatriciaTree.hs

```diff
@@ -220,6 +224,25 @@ toAdj = concatMap expand . IM.toList
 fromAdj :: Adj b -> IntMap [b]
 fromAdj = IM.fromListWith addLists . map (second (:[]) . swap)
 
+data FromListCounting a = FromListCounting !Int !(IntMap a)
```

Comment:
I've left the instances out for now; if you can think of a reason to add them, I won't object, but I don't think it serves any purpose right now except to add compilation time and bloat the compiler output. It's just a throwaway datatype; I tend to use loads of them.